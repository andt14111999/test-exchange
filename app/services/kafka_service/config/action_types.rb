# frozen_string_literal: true

module KafkaService
  module Config
    module ActionTypes
      # Transaction types
      COIN_TRANSACTION = 'CoinTransaction'

      # Account types
      COIN_ACCOUNT = 'CoinAccount'
      FIAT_ACCOUNT = 'FiatAccount'

      # Merchant types
      MERCHANT_ESCROW = 'MerchantEscrow'
    end
  end
end
