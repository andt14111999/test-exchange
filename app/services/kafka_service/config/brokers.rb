# frozen_string_literal: true

module KafkaService
  module Config
    module Brokers
      BROKERS = [ ENV.fetch('KAFKA_BROKER') || '127.0.0.1:9092' ].freeze
    end

    module Config
      CONSUMER_CONFIG = {
        offset_commit_interval: 5,
        offset_commit_threshold: 100,
        offset_retention_time: 7_200,
        fetcher_max_queue_size: 100,
        session_timeout: 30,
        heartbeat_interval: 10
      }.freeze

      PRODUCER_CONFIG = {
        required_acks: :all,
        compression_codec: :snappy,
        max_retries: 3,
        retry_backoff: 1,
        max_buffer_size: 10_000,
        max_buffer_bytesize: 10_485_760
      }.freeze
    end
  end
end
