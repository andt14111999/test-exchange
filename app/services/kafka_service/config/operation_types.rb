# frozen_string_literal: true

module KafkaService
  module Config
    module OperationTypes
      # Coin operations
      COIN_DEPOSIT_CREATE = 'coin_deposit_create'
      COIN_WITHDRAWAL_CREATE = 'coin_withdrawal_create'
      COIN_WITHDRAWAL_RELEASING = 'coin_withdrawal_releasing'
      COIN_WITHDRAWAL_FAILED = 'coin_withdrawal_failed'
      COIN_WITHDRAWAL_CANCELLED = 'coin_withdrawal_cancelled'
      COIN_ACCOUNT_CREATE = 'coin_account_create'
      COIN_ACCOUNT_QUERY_BALANCE = 'coin_account_query_balance'
      COIN_ACCOUNT_RESET_BALANCE = 'coin_account_reset_balance'

      # AMM operations
      AMM_POOL_CREATE = 'amm_pool_create'
      AMM_POOL_UPDATE = 'amm_pool_update'

      # Merchant operations
      MERCHANT_ESCROW_MINT = 'merchant_escrow_mint'
      MERCHANT_ESCROW_BURN = 'merchant_escrow_burn'

      # Legacy
      BALANCE_QUERY = 'balance_query'
    end
  end
end
