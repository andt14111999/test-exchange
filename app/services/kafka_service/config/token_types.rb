# frozen_string_literal: true

module KafkaService
  module Config
    module TokenTypes
      USDT = 'USDT'
      BTC = 'BTC'
      ETH = 'ETH'
      SOL = 'SOL'
    end
  end
end
