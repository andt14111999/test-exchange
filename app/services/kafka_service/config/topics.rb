# frozen_string_literal: true

module KafkaService
  module Config
    module Topics
      # Input topics
      COIN_ACCOUNT = 'EE.I.coin_account'
      COIN_DEPOSIT = 'EE.I.coin_deposit'
      COIN_WITHDRAW = 'EE.I.coin_withdraw'
      COIN_ACCOUNT_QUERY = 'EE.I.coin_account_query'
      COIN_ACCOUNT_RESET = 'EE.I.coin_account_reset'

      # Output topics
      COIN_ACCOUNT_UPDATE = 'EE.O.coin_account_update'
      TRANSACTION_RESPONSE = 'EE.O.transaction_response'

      # Aliases
      BALANCE_UPDATE = COIN_ACCOUNT_UPDATE
      TRANSACTION_RESULT = TRANSACTION_RESPONSE
      BALANCE_RESPONSE = COIN_ACCOUNT_UPDATE
    end
  end
end
