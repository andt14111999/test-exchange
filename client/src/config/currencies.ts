export interface CurrencyInfo {
  name: string;
  symbol: string;
  type: "crypto" | "fiat";
  icon?: string;
}

export const CURRENCIES: Record<string, CurrencyInfo> = {
  // Cryptocurrencies
  BTC: {
    name: "Bitcoin",
    symbol: "BTC",
    type: "crypto",
    icon: "₿",
  },
  ETH: {
    name: "Ethereum",
    symbol: "ETH",
    type: "crypto",
    icon: "Ξ",
  },
  BNB: {
    name: "BNB",
    symbol: "BNB",
    type: "crypto",
    icon: "BNB",
  },
  USDT: {
    name: "Tether",
    symbol: "USDT",
    type: "crypto",
    icon: "USDT",
  },
  VND: {
    name: "Vietnamese Dong",
    symbol: "VND",
    type: "fiat",
    icon: "₫",
  },
  PHP: {
    name: "Philippine Peso",
    symbol: "PHP",
    type: "fiat",
    icon: "₱",
  },
};
