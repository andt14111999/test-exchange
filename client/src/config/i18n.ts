import { getRequestConfig } from "next-intl/server";

export const locales = ["en", "vi", "fil"] as const;
export const defaultLocale = "en" as const;

export type Locale = (typeof locales)[number];

export default getRequestConfig(async ({ locale }) => {
  const resolvedLocale =
    !locale || !locales.includes(locale as Locale)
      ? defaultLocale
      : (locale as string);

  return {
    messages: (await import(`@/i18n/${resolvedLocale}.json`)).default,
    locale: resolvedLocale,
    timeZone: "Asia/Ho_Chi_Minh",
  };
});
