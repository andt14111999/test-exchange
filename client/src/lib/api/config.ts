export const API_BASE_URL =
  process.env.NEXT_PUBLIC_API_BASE_URL || "http://localhost:3969/api/v1";

export const WS_URL = process.env.NEXT_PUBLIC_WS_URL || "ws://localhost:3969";

export const API_ENDPOINTS = {
  auth: {
    google: "/auth/google",
  },
  users: {
    me: "/users/me",
    updateUsername: "/users/username",
  },
  wallet: {
    balances: "/balances",
    transactions: "/coin_transactions",
  },
  fiat: {
    deposits: "/fiat_deposits",
    withdrawals: "/fiat_withdrawals",
    deposit: {
      create: "/fiat_deposits",
      createForTrade: "/fiat_deposits/for_trade",
      get: (id: string) => `/fiat_deposits/${id}`,
      markMoneySent: (id: string) => `/fiat_deposits/${id}/money_sent`,
      verifyOwnership: (id: string) => `/fiat_deposits/${id}/verify_ownership`,
      cancel: (id: string) => `/fiat_deposits/${id}/cancel`,
    },
    withdrawal: {
      create: "/fiat_withdrawals",
      get: (id: string) => `/fiat_withdrawals/${id}`,
      markMoneySent: (id: string) => `/fiat_withdrawals/${id}/money_sent`,
      verifyOwnership: (id: string) =>
        `/fiat_withdrawals/${id}/verify_ownership`,
      cancel: (id: string) => `/fiat_withdrawals/${id}/cancel`,
    },
  },
  transactions: {
    list: "/transactions",
    get: (id: string) => `/transactions/${id}`,
  },
  coinAccounts: {
    address: (coinCurrency: string, layer: string) =>
      `/coin_accounts/address?coin_currency=${coinCurrency}&layer=${layer}`,
    generateAddress: (coinCurrency: string, layer: string) =>
      `/coin_accounts/generate_address?coin_currency=${coinCurrency}&layer=${layer}`,
  },
  merchant: {
    register: "/merchant_registration",
    escrows: {
      list: "/merchant_escrows",
      create: "/merchant_escrows",
      get: (id: number) => `/merchant_escrows/${id}`,
      cancel: (id: number) => `/merchant_escrows/${id}/cancel`,
    },
    offers: {
      list: "/offers",
      merchantList: "/offers/merchant",
      get: (id: number) => `/offers/${id}`,
      create: "/offers",
      update: (id: number) => `/offers/${id}`,
      delete: (id: number) => `/offers/${id}`,
      enable: (id: number) => `/offers/${id}/enable`,
      disable: (id: number) => `/offers/${id}/disable`,
      setOnlineStatus: (id: number) => `/offers/${id}/online_status`,
    },
  },
  trades: {
    list: "/trades",
    get: (id: string) => `/trades/${id}`,
    create: "/trades",
    markPaid: (id: string) => `/trades/${id}/mark_paid`,
    release: (id: string) => `/trades/${id}/complete`,
    dispute: (id: string) => `/trades/${id}/dispute`,
    cancel: (id: string) => `/trades/${id}/cancel`,
    fiatDeposit: {
      create: (tradeId: string) => `/trades/${tradeId}/fiat_deposit`,
      get: (tradeId: string) => `/trades/${tradeId}/fiat_deposit`,
      markMoneySent: (tradeId: string) =>
        `/trades/${tradeId}/fiat_deposit/money_sent`,
      verifyOwnership: (tradeId: string) =>
        `/trades/${tradeId}/fiat_deposit/verify_ownership`,
      cancel: (tradeId: string) => `/trades/${tradeId}/fiat_deposit/cancel`,
    },
  },
  amm: {
    pools: "/amm_pools",
    activePools: "/amm_pools/active",
    poolDetail: (pair: string) => `/amm_pools/${pair}`,
    positions: "/amm_positions",
    positionDetail: (id: number) => `/amm_positions/${id}`,
    collectFee: (id: number) => `/amm_positions/${id}/collect_fee`,
    closePosition: (id: number) => `/amm_positions/${id}/close`,
    orders: "/amm_orders",
    ticks: "/ticks",
  },
  withdrawals: {
    create: "/coin_withdrawals",
    get: (id: string) => `/coin_withdrawals/${id}`,
  },
  coins: "/coins",
  banks: "/banks",
  bankAccounts: {
    list: "/bank_accounts",
    create: "/bank_accounts",
    update: (id: string) => `/bank_accounts/${id}`,
    delete: (id: string) => `/bank_accounts/${id}`,
    setPrimary: (id: string) => `/bank_accounts/${id}/set_primary`,
  },
  paymentMethods: {
    list: "/payment_methods",
    get: (id: number) => `/payment_methods/${id}`,
  },
  settings: {
    coinSettings: "/coin_settings",
  },
} as const;
