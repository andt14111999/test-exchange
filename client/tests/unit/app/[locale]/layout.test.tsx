import React from "react";
import { render } from "@testing-library/react";
import LocaleLayout from "@/app/[locale]/layout";
import { getMessages } from "next-intl/server";
import { defaultLocale } from "@/config/i18n";

// Mock next-intl
jest.mock("next-intl/server", () => ({
  getMessages: jest.fn(),
  getRequestConfig: jest.fn(),
}));

// Mock next-intl client provider
jest.mock("next-intl", () => ({
  NextIntlClientProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="next-intl-provider">{children}</div>
  ),
}));

// Mock all providers
jest.mock("@/components/providers/number-format-provider", () => ({
  NumberFormatProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="number-format-provider">{children}</div>
  ),
}));

jest.mock("@/components/providers/query-provider", () => ({
  QueryProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="query-provider">{children}</div>
  ),
}));

jest.mock("@/components/providers/client-pool-initializer", () => ({
  ClientPoolInitializer: () => <div data-testid="client-pool-initializer" />,
}));

jest.mock("@/components/providers/notification-provider", () => ({
  NotificationProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="notification-provider">{children}</div>
  ),
}));

jest.mock("@/components/providers/balance-provider", () => ({
  BalanceProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="balance-provider">{children}</div>
  ),
}));

jest.mock("@/components/layout/header", () => ({
  Header: () => <div data-testid="header" />,
}));

jest.mock("@/components/ui/toast", () => ({
  Toast: () => <div data-testid="toast" />,
}));

describe("LocaleLayout", () => {
  const mockMessages = { test: "test" };
  const mockChildren = <div data-testid="test-children">Test Content</div>;

  beforeEach(() => {
    jest.clearAllMocks();
    (getMessages as jest.Mock).mockResolvedValue(mockMessages);
  });

  it("should render with valid locale", async () => {
    const params = Promise.resolve({ locale: "en" });
    const { container } = render(
      await LocaleLayout({ children: mockChildren, params }),
    );

    expect(getMessages).toHaveBeenCalledWith({ locale: "en" });
    expect(container).toMatchSnapshot();
  });

  it("should use default locale when locale is invalid", async () => {
    const params = Promise.resolve({ locale: "invalid" });
    const { container } = render(
      await LocaleLayout({ children: mockChildren, params }),
    );

    expect(getMessages).toHaveBeenCalledWith({ locale: defaultLocale });
    expect(container).toMatchSnapshot();
  });

  it("should use default locale when locale is empty", async () => {
    const params = Promise.resolve({ locale: "" });
    const { container } = render(
      await LocaleLayout({ children: mockChildren, params }),
    );

    expect(getMessages).toHaveBeenCalledWith({ locale: defaultLocale });
    expect(container).toMatchSnapshot();
  });

  it("should render all required components in correct order", async () => {
    const params = Promise.resolve({ locale: "en" });
    const { getByTestId } = render(
      await LocaleLayout({ children: mockChildren, params }),
    );

    expect(getByTestId("next-intl-provider")).toBeInTheDocument();
    expect(getByTestId("number-format-provider")).toBeInTheDocument();
    expect(getByTestId("query-provider")).toBeInTheDocument();
    expect(getByTestId("client-pool-initializer")).toBeInTheDocument();
    expect(getByTestId("notification-provider")).toBeInTheDocument();
    expect(getByTestId("balance-provider")).toBeInTheDocument();
    expect(getByTestId("header")).toBeInTheDocument();
    expect(getByTestId("toast")).toBeInTheDocument();
    expect(getByTestId("test-children")).toBeInTheDocument();
  });
});
