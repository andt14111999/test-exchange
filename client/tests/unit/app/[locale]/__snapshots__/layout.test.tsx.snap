// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleLayout should render with valid locale 1`] = `
<div>
  <div
    data-testid="next-intl-provider"
  >
    <div
      data-testid="number-format-provider"
    >
      <div
        data-testid="query-provider"
      >
        <div
          data-testid="client-pool-initializer"
        />
        <div
          data-testid="notification-provider"
        >
          <div
            data-testid="balance-provider"
          >
            <div
              class="relative min-h-screen flex flex-col"
            >
              <div
                data-testid="header"
              />
              <main
                class="flex-1"
              >
                <div
                  class="container mx-auto max-w-7xl px-4 sm:px-6 lg:px-8 py-6"
                >
                  <div
                    data-testid="test-children"
                  >
                    Test Content
                  </div>
                </div>
              </main>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="toast"
    />
  </div>
</div>
`;

exports[`LocaleLayout should use default locale when locale is empty 1`] = `
<div>
  <div
    data-testid="next-intl-provider"
  >
    <div
      data-testid="number-format-provider"
    >
      <div
        data-testid="query-provider"
      >
        <div
          data-testid="client-pool-initializer"
        />
        <div
          data-testid="notification-provider"
        >
          <div
            data-testid="balance-provider"
          >
            <div
              class="relative min-h-screen flex flex-col"
            >
              <div
                data-testid="header"
              />
              <main
                class="flex-1"
              >
                <div
                  class="container mx-auto max-w-7xl px-4 sm:px-6 lg:px-8 py-6"
                >
                  <div
                    data-testid="test-children"
                  >
                    Test Content
                  </div>
                </div>
              </main>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="toast"
    />
  </div>
</div>
`;

exports[`LocaleLayout should use default locale when locale is invalid 1`] = `
<div>
  <div
    data-testid="next-intl-provider"
  >
    <div
      data-testid="number-format-provider"
    >
      <div
        data-testid="query-provider"
      >
        <div
          data-testid="client-pool-initializer"
        />
        <div
          data-testid="notification-provider"
        >
          <div
            data-testid="balance-provider"
          >
            <div
              class="relative min-h-screen flex flex-col"
            >
              <div
                data-testid="header"
              />
              <main
                class="flex-1"
              >
                <div
                  class="container mx-auto max-w-7xl px-4 sm:px-6 lg:px-8 py-6"
                >
                  <div
                    data-testid="test-children"
                  >
                    Test Content
                  </div>
                </div>
              </main>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="toast"
    />
  </div>
</div>
`;
